### ISSUE TITLE
Crazy long i3 booting up time after recent merge

### ISSUE THREAD

**Issue by sarpik** [LukeSmithxyz/voidrice#430]
I've just finished merging [the latest changes](https://github.com/sarpik/voidrice) - everything seems fine.

Except that now, once I log in, I just get an empty i3 screen with my background image, and have to wait around 2 minutes for everything to start working - the statusbar to show up, to get the keyboard shortcuts do anything etc.

This started happening just now, right after finishing my latest merge and rebooting.

If I checkout my dotfiles to [the latest commit before merging](https://github.com/sarpik/voidrice/tree/2429f7606303b4c2fd27957b2f0ee3c1018a0575) 
and re-login - the issue disappears.

Similarly - switching back to the latest changes + re-logging in *still* works fine, but after rebooting, the issue comes back.

Has anyone here possibly encountered something like this and maybe could help me out?

Thanks!

---

P.S.

Once I launch `st`, sometimes it takes 5x the time, like instead of the normal `50`ms, it takes `250`ms. Thus it might relate to this issue, since it started happening at the same time.

Could the `$ZDOTDIR` have any impact here? Previously I just didn't have it & my `.zshrc` was located in `$HOME`.

Edit:

* Disabling `$ZDOTDIR` / removing `~/.config/zsh/.zshrc` or `~/.config/aliasrc` did not help.

* I've tried bisecting, but since the i3 configs weren't always available, I was not able to test if the issue persists.

```sh
git bisect start HEAD 2429f7606303b4c2fd27957b2f0ee3c1018a0575

# test if the issue still persists
git bisect good|bad
# repeat

git bisect reset # once done
```

---

Edit: After the [recent fixes with colorful emojis regarding st / dmenu (xorg)](https://youtu.be/f9qNXV01yzg), updating to the even more latest changes would be extremely useful.

**Comment by Daja177** [comment]
@sarpik I'm not surprised you're experiencing issues after the merge. The merge 014c91897dc8f00fa638289361e2453b7addd1c4 has a **huge** amount of changes that will definitely conflict with your current setup if you don't make a lot of manual changes, such as switching from `pulseaudio` to `alsa` and a lot more.

I'd recommend that you revert the merge if you're still experiencing these slowdowns, and instead go through the changes made on Luke's branch and manually add the changes you want.

**Comment by sarpik** [comment]
@Daja177 hey there, thanks for your input!

oh boy, yes it indeed does have lots of changes.

TL;DR:

Starting all over again - naaah. It might even be worse, because I might miss something important etc., but I don't mind making more manual changes if they'd help solve my current situation with the master branch.

---

I've kept the merge at master, but jumped back at the pre-merge commit, created [the `current` branch](https://github.com/sarpik/voidrice/tree/current) & am currently using + committing new changes to it.

Going through Luke's changes 1 by 1 is kinda too much effort for what I'd get - there were about 200 commits from Luke, and 300+ from me.

The merge itself, obviously, had lots of conflicts, and I don't really want to re-do the same stuff again:D (and if cherry-picking / rebasing - even more work!).

I thought that maybe there was some obvious solution or a fix that I wasn't aware of that would help me solve the issue, but so far, nothing has helped.

**Comment by Daja177** [comment]
When I suggested adding the changes manually I didn't necessarily mean picking the specific commits, but I see why you'd want to do that haha

Looking through all of the changes it seems most are simply bloat cleanup, reformatting, reorganization etc. The most important changes I find is the addition of config files for `lf`, `ranger`, `bspwm` and `zsh`, along with the replacement of `pulseaudio` commands to `alsa` ones in scripts and config files.

Unfortunately I'm unable to pinpoint whatever is causing the slowdowns even after looking through it several times (although I wouldn't be surprised if it was *more* than one factor), but it doesn't seem like you're missing out on anything special by sticking to [the `current` branch](https://github.com/sarpik/voidrice/tree/current). But if you want to add Luke's `lf`, `ranger`, `bspwm` or `zsh` config files you should easily be able to find the correct commits to merge.

**Comment by vladdoster** [comment]
You could try this, what do you think @Daja177 ?

```
exec i3 -V >> ~/i3log-$(date +'%F-%k-%M-%S') 2>&1
```

**Comment by sarpik** [comment]
@Daja177 lol
Yeah I haven't found any "strong" changes per se.

Though I would really appreciate having them - I currently have some issues with gnupg / gpg keys not being automatically unlocked (something to do with the .password-store, keyrings etc., it's a rabbit hole) and lots of other minor nit pics that were solved after I merged in the changes. (sadly, as mentioned - I can't use them - once I reboot, the start-up times are atrocious).

I feel like it's either all or nothing at this point - I'm fine without the minor changes, but I would really like having 'em all!:D

@vladdoster hey there, thanks.
Also, I think you meant to use `%H` instead of `%k`, since `%k` adds a space & messes the thing up:D

```sh
exec i3 -V >> ~/i3log-$(date +'%F-%H-%M-%S') 2>&1
```

I don't know if I mentioned this, but I tried `git bisect`ing while trying to measure the start-up times for i3.
I failed, because on some commits, `i3` was not even installed (or it wasn't in the `$PATH`) - either way, I didn't knew how to reach i3, and thus couldn't test out & find the broken commit.
I think I should retry doing this to see if I can get i3 somehow & find the broken commit.

Also, if the start-up time is bad, you don't need a script to measure it - it's very, *very* obvious (compare 2 seconds to ~ 3-4 minutes:D)

**Comment by vladdoster** [comment]
@sarpik no, I mean %k. Which gives the hour but space padded. Same as %_H.

**Comment by sarpik** [comment]
@vladdoster ah I thought it was a mistake since the added space makes is not picked up once you pipe the output of `exec i3 -V`, thus piping it to the same file every time.
Ah, I just noticed that you're using `>>` so I assume you meant to append to it - all cool, just don't know why you'd have the `-%M-%S'` after `%k` since it doesn't get picked up. I'm just curious:D & thanks anyway

**Comment by vladdoster** [comment]
@sarpik yes, the intended use is to create a single log file. I dont see what you mean by it not being picked up? Seems right to me... it is January 30th, 2020 at 17:52:38? Am I missing something?
![pic-selected-200130-1756-58](https://user-images.githubusercontent.com/10052309/73497941-f64c5680-4389-11ea-96e3-da2c8905c84e.png)

**Comment by sarpik** [comment]
@vladdoster yeah I understand it now - you meant to echo the time into the file itself, not create a separate file every time. Thanks.

Anyhow, any other ideas to solve the issue?

**Comment by vladdoster** [comment]
@sarpik exactly! 

I'd probably run the logging and try to see if there are any big time chunks that you can mail down to a program. Other than that, I am just as lost :(.

**Comment by sarpik** [comment]
@vladdoster alright cool, thanks for your input:)

**Comment by LukeSmithxyz** [comment]
I don't know if you're still trying to figure this out, but might it have been solved by https://github.com/LukeSmithxyz/voidrice/pull/446?

**Comment by sarpik** [comment]
@LukeSmithxyz 

> I don't know if you're still trying to figure this out, but might it have been solved by #446?

Last time I checked, I still had issues. Thanks for linking the #446 - will try out soon & report back here!

**Comment by sarpik** [comment]
Aight so I've been running the new setup for a few weeks.

The boot-up time is now acceptable - only a few seconds.

But the overall laggy-ness is still way worse, compared to the setup without latest changes from here. I'm sadly forced to stick to the old setup, because everything takes just too much time & it's too annoying, sadly:/

**Comment by LukeSmithxyz** [comment]
What exactly is "laggy?" Opening windows or something? I'm curious if anyone else is having these problems or if they're specific to your machine.

**Comment by sarpik** [comment]
@LukeSmithxyz

> What exactly is "laggy?"

Launching apps, such as `st`, `dmenu`, `chrome`, `vscode`, even `nvim` from `st` - all of them take more time than usual & it's slowing down my workflow. Even `zsh` starts up pretty slow, even though there aren't any changes introduced when comparing the setups.

It might be specific to my machine, since on my laptop I have windows, and I dual-boot arch via my USB (up to 420 MB/s read, 380 MB/s write) - it's normal that the I/O speeds are not as high as, say, an SSD, but once again - previously I didn't have any issues and everything was fast enough, but after the latest changes - not anymore.

**Comment by vladdoster** [comment]
Have you installed the Intel microcode?

**Comment by sarpik** [comment]
@vladdoster 
nope, at least not explicitly. Looking it up right now. Has it anything to do with performance? I only see the security patches, which're important, but is there anything else I need to know?

**Comment by sarpik** [comment]
@vladdoster 
Aight I've installed the updates, including `Intel microcode`.
After testing the new arch setup - still no improvements, sadly:/

**Comment by LukeSmithxyz** [comment]
I mean could you just give us a screenshot of htop or some other system monitor?

Are you making fresh installs or are you updating a preexisting user/machine?

**Comment by vladdoster** [comment]
@LukeSmithxyz yeah we've been shooting in the dark here. Good idea

**Comment by sarpik** [comment]
Sure. Just tell me what ya'll need:D

First I ran everything with my normal setup - the one which works relatively fine. Then rebooted and  did the same with the slower / laggy setup - it was noticeably slower, as expected.

Both times, I ran `ps aux`, `ps auxf` and `htop -d 100000000000` to avoid any updates. I ran it right after rebooting & waiting a bit. I noticed a loooot of duplicates or similar processes - is this normal?

There was a lot of stuff, so I just threw everything into google drive here:
https://drive.google.com/open?id=11Zzv__Fgv5ZZk3gajAps3I4DmXbOFmaD

Anything else I can help with?

**Comment by sarpik** [comment]
@LukeSmithxyz 
> Are you making fresh installs or are you updating a preexisting user/machine?

I'm not doing anything with the main machine, which runs windows. I have a flash USB drive, and inside it I have arch linux installed. I have not re-installed, I'm doing everything inside the same USB, the same installation of arch.

I'm switching between the two setups by simply checking out a different branch in git and rebooting.

**Comment by vladdoster** [comment]
Mhm... I've personally never heard of arch installed on a USB drive. So I'm not of much help.

@sarpik Why do you have everything commented out in .Xprofile? Are you sure the dbus thing is t causing issues?

**Comment by vladdoster** [comment]
You might also try to do a recursive diff to see what the differences are? Clone each branch into a seperate folder and the pass both to diff -r.

**Comment by vladdoster** [comment]
#!/bin/bash
# Skip files in $1 which are symlinks
for f in `find $1/* ! -type l`
do
    # Suppress details of differences
    diff -rq $f $2/${f##*/}
done

**Comment by sarpik** [comment]
> @sarpik Why do you have everything commented out in .Xprofile?

I have everything commented out in the laggy setup, because if I don't - instead of a laggy one, it becomes completely unusable - it would take 3+ minutes to just load i3.
I haven't checked which part of the .Xprofile would cause the huge spike in load time though.

> Are you sure the dbus thing is t causing issues?

From `git blame`, I see that the dbus thing was introduced not by me, but by Luke:

```
06d379ce (Luke Smith        2020-02-09 07:26:32 -0500  6) # Fix Gnome Apps Slow  Start due to failing services
06d379ce (Luke Smith        2020-02-09 07:26:32 -0500  7) # Add this when you include flatpak in your system
06d379ce (Luke Smith        2020-02-09 07:26:32 -0500  8) dbus-update-activation-environment --systemd DBUS_SESSION_BUS_ADDRESS DISPLAY XAUTHORITY
```

in the `current` setup (not laggy), I do not have it - it might be a problem - I'll check soon, but I doubt it.


> You might also try to do a recursive diff to see what the differences are? Clone each branch into a seperate folder and the pass both to diff -r.

I tried doing this with the script you provided like so:

```sh
$ ./diffit voidrice-master-laggy voidrice-current-okay
```

but it somehow only outputs this:

```console
Files voidrice-master-laggy/README.md and voidrice-current-okay/README.md differ
```

---

I then tried running this:

```sh
diff -rq voidrice-master-laggy voidrice-current-okay
```

and got

```console
Only in voidrice-master-laggy: .asoundrc
Only in voidrice-current-okay: .bash_profile
Files voidrice-master-laggy/.bashrc.off and voidrice-current-okay/.bashrc.off differ
Files voidrice-master-laggy/.config/aliasrc and voidrice-current-okay/.config/aliasrc differ
Only in voidrice-current-okay/.config: bmdirs
Only in voidrice-current-okay/.config: bmfiles
Only in voidrice-master-laggy/.config: bspwm
Only in voidrice-master-laggy/.config: directories
Only in voidrice-current-okay/.config: emoji
Only in voidrice-master-laggy/.config: files
Only in voidrice-current-okay/.config: fontawesome
Files voidrice-master-laggy/.config/fontconfig/fonts.conf and voidrice-current-okay/.config/fontconfig/fonts.conf differ
Only in voidrice-current-okay/.config/getkeys: calcurse
Only in voidrice-current-okay/.config/getkeys: i3
Only in voidrice-current-okay/.config/getkeys: mutt
Only in voidrice-current-okay/.config/getkeys: ncmpcpp
Only in voidrice-current-okay/.config/getkeys: newsboat
Only in voidrice-current-okay/.config/getkeys: sxiv
Only in voidrice-current-okay/.config/getkeys: zathura
Files voidrice-master-laggy/.config/gtk-2.0/gtkrc-2.0 and voidrice-current-okay/.config/gtk-2.0/gtkrc-2.0 differ
Files voidrice-master-laggy/.config/i3/config and voidrice-current-okay/.config/i3/config differ
Files voidrice-master-laggy/.config/i3blocks/config and voidrice-current-okay/.config/i3blocks/config differ
Only in voidrice-master-laggy/.config: inputrc
Only in voidrice-master-laggy/.config: lf
Files voidrice-master-laggy/.config/mimeapps.list and voidrice-current-okay/.config/mimeapps.list differ
Files voidrice-master-laggy/.config/mpd/mpd.conf and voidrice-current-okay/.config/mpd/mpd.conf differ
Files voidrice-master-laggy/.config/mpv/input.conf and voidrice-current-okay/.config/mpv/input.conf differ
Files voidrice-master-laggy/.config/ncmpcpp/bindings and voidrice-current-okay/.config/ncmpcpp/bindings differ
Files voidrice-master-laggy/.config/ncmpcpp/config and voidrice-current-okay/.config/ncmpcpp/config differ
Files voidrice-master-laggy/.config/nvim/init.vim and voidrice-current-okay/.config/nvim/init.vim differ
Only in voidrice-master-laggy/.config: polybar
Only in voidrice-master-laggy/.config/ranger: commands.py
Only in voidrice-current-okay/.config/ranger: luke_ranger_readme.md
Files voidrice-master-laggy/.config/ranger/rc.conf and voidrice-current-okay/.config/ranger/rc.conf differ
Only in voidrice-current-okay/.config/ranger: rifle.conf
Only in voidrice-current-okay/.config/ranger: scope.sh
Only in voidrice-current-okay/.config: redshift
Files voidrice-master-laggy/.config/sxhkd/sxhkdrc and voidrice-current-okay/.config/sxhkd/sxhkdrc differ
Files voidrice-master-laggy/.config/sxiv/exec/key-handler and voidrice-current-okay/.config/sxiv/exec/key-handler differ
Only in voidrice-master-laggy/.config: wal
Only in voidrice-master-laggy/.config: wget
Only in voidrice-master-laggy/.config: Xresources
Only in voidrice-master-laggy/.config/zsh: .zshrc
Files voidrice-master-laggy/.git/config and voidrice-current-okay/.git/config differ
Files voidrice-master-laggy/.git/HEAD and voidrice-current-okay/.git/HEAD differ
Files voidrice-master-laggy/.git/index and voidrice-current-okay/.git/index differ
Files voidrice-master-laggy/.git/logs/HEAD and voidrice-current-okay/.git/logs/HEAD differ
Only in voidrice-current-okay/.git/logs/refs/heads: current
Only in voidrice-current-okay/.git/refs/heads: current
Only in voidrice-current-okay: .icons
Only in voidrice-current-okay: .inputrc
Only in voidrice-master-laggy/.local/bin: basic-chrome-profile
Only in voidrice-master-laggy/.local/bin: chrome-profile.js
Only in voidrice-master-laggy/.local/bin: compiler
Files voidrice-master-laggy/.local/bin/cron/checkup and voidrice-current-okay/.local/bin/cron/checkup differ
Files voidrice-master-laggy/.local/bin/cron/cronbat and voidrice-current-okay/.local/bin/cron/cronbat differ
Files voidrice-master-laggy/.local/bin/cron/crontog and voidrice-current-okay/.local/bin/cron/crontog differ
Files voidrice-master-laggy/.local/bin/cron/IMPORTANT_NOTE.md and voidrice-current-okay/.local/bin/cron/IMPORTANT_NOTE.md differ
Files voidrice-master-laggy/.local/bin/cron/newsup and voidrice-current-okay/.local/bin/cron/newsup differ
Only in voidrice-master-laggy/.local/bin: dif
Only in voidrice-master-laggy/.local/bin: displayselect
Only in voidrice-master-laggy/.local/bin: dmenuhandler
Only in voidrice-master-laggy/.local/bin: dmenumount
Only in voidrice-master-laggy/.local/bin: dmenupass
Only in voidrice-master-laggy/.local/bin: dmenurecord
Only in voidrice-master-laggy/.local/bin: dmenuumount
Only in voidrice-master-laggy/.local/bin: dmenuunicode
Only in voidrice-master-laggy/.local/bin: ecryptfs-unmount
Only in voidrice-master-laggy/.local/bin: ext
Only in voidrice-master-laggy/.local/bin: getkeys
Only in voidrice-master-laggy/.local/bin: git-behind
Only in voidrice-master-laggy/.local/bin: git-rm-backup
Only in voidrice-master-laggy/.local/bin: git-wrapper
Only in voidrice-master-laggy/.local/bin: hide
Only in voidrice-current-okay/.local/bin/i3cmds: cabl
Only in voidrice-current-okay/.local/bin/i3cmds: camtoggle
Files voidrice-master-laggy/.local/bin/i3cmds/ddspawn and voidrice-current-okay/.local/bin/i3cmds/ddspawn differ
Only in voidrice-current-okay/.local/bin/i3cmds: displayselect
Only in voidrice-current-okay/.local/bin/i3cmds: dmenumount
Only in voidrice-current-okay/.local/bin/i3cmds: dmenurecord
Only in voidrice-current-okay/.local/bin/i3cmds: dmenuumount
Only in voidrice-current-okay/.local/bin/i3cmds: dmenuunicode
Files voidrice-master-laggy/.local/bin/i3cmds/dropdowncalc and voidrice-current-okay/.local/bin/i3cmds/dropdowncalc differ
Only in voidrice-current-okay/.local/bin/i3cmds: ducksearch
Files voidrice-master-laggy/.local/bin/i3cmds/hover and voidrice-current-okay/.local/bin/i3cmds/hover differ
Files voidrice-master-laggy/.local/bin/i3cmds/i3resize and voidrice-current-okay/.local/bin/i3cmds/i3resize differ
Only in voidrice-current-okay/.local/bin/i3cmds: killrecording
Only in voidrice-current-okay/.local/bin/i3cmds: maimpick
Only in voidrice-current-okay/.local/bin/i3cmds: prompt
Only in voidrice-current-okay/.local/bin/i3cmds: samedir
Only in voidrice-current-okay/.local/bin/i3cmds: showclip
Only in voidrice-current-okay/.local/bin/i3cmds: td-toggle
Files voidrice-master-laggy/.local/bin/i3cmds/tmuxdd and voidrice-current-okay/.local/bin/i3cmds/tmuxdd differ
Only in voidrice-current-okay/.local/bin/i3cmds: toggletouchpad
Only in voidrice-current-okay/.local/bin/i3cmds: torwrap
Only in voidrice-current-okay/.local/bin/i3cmds: tutorialvids
Only in voidrice-current-okay/.local/bin/i3cmds: winresize
Only in voidrice-master-laggy/.local/bin: i3lockstrong
Only in voidrice-master-laggy/.local/bin: ifinstalled
Only in voidrice-master-laggy/.local/bin: keyboard-remaps
Only in voidrice-master-laggy/.local/bin: launch_polybar
Only in voidrice-master-laggy/.local/bin: lf-select
Only in voidrice-master-laggy/.local/bin: linkhandler
Only in voidrice-master-laggy/.local/bin: maimpick
Only in voidrice-master-laggy/.local/bin: mpd-module-update
Only in voidrice-master-laggy/.local/bin: notify-break-1
Only in voidrice-master-laggy/.local/bin: notify-break-5
Only in voidrice-master-laggy/.local/bin: opout
Only in voidrice-master-laggy/.local/bin: pauseallmpv
Only in voidrice-master-laggy/.local/bin: podentr
Only in voidrice-master-laggy/.local/bin: prompt
Only in voidrice-master-laggy/.local/bin: qndl
Only in voidrice-master-laggy/.local/bin: queueandnotify
Only in voidrice-master-laggy/.local/bin: refbar
Only in voidrice-master-laggy/.local/bin: remaps
Only in voidrice-master-laggy/.local/bin: restart-wifi
Only in voidrice-master-laggy/.local/bin: rotdir
Only in voidrice-master-laggy/.local/bin: rssadd
Only in voidrice-master-laggy/.local/bin: samedir
Only in voidrice-master-laggy/.local/bin: screenlock
Only in voidrice-current-okay/.local/bin: SCRIPTS.md
Only in voidrice-master-laggy/.local/bin: setbg
Only in voidrice-master-laggy/.local/bin: shortcuts
Only in voidrice-master-laggy/.local/bin: showclip
Files voidrice-master-laggy/.local/bin/statusbar/battery and voidrice-current-okay/.local/bin/statusbar/battery differ
Files voidrice-master-laggy/.local/bin/statusbar/clock and voidrice-current-okay/.local/bin/statusbar/clock differ
Files voidrice-master-laggy/.local/bin/statusbar/disk and voidrice-current-okay/.local/bin/statusbar/disk differ
Files voidrice-master-laggy/.local/bin/statusbar/help and voidrice-current-okay/.local/bin/statusbar/help differ
Files voidrice-master-laggy/.local/bin/statusbar/internet and voidrice-current-okay/.local/bin/statusbar/internet differ
Files voidrice-master-laggy/.local/bin/statusbar/iplocate and voidrice-current-okay/.local/bin/statusbar/iplocate differ
Files voidrice-master-laggy/.local/bin/statusbar/mailbox and voidrice-current-okay/.local/bin/statusbar/mailbox differ
Only in voidrice-master-laggy/.local/bin/statusbar: moonphase
Only in voidrice-current-okay/.local/bin/statusbar: mpdupdate
Files voidrice-master-laggy/.local/bin/statusbar/music and voidrice-current-okay/.local/bin/statusbar/music differ
Files voidrice-master-laggy/.local/bin/statusbar/news and voidrice-current-okay/.local/bin/statusbar/news differ
Files voidrice-master-laggy/.local/bin/statusbar/pacpackages and voidrice-current-okay/.local/bin/statusbar/pacpackages differ
Files voidrice-master-laggy/.local/bin/statusbar/popupgrade and voidrice-current-okay/.local/bin/statusbar/popupgrade differ
Files voidrice-master-laggy/.local/bin/statusbar/torrent and voidrice-current-okay/.local/bin/statusbar/torrent differ
Files voidrice-master-laggy/.local/bin/statusbar/volume and voidrice-current-okay/.local/bin/statusbar/volume differ
Files voidrice-master-laggy/.local/bin/statusbar/weather and voidrice-current-okay/.local/bin/statusbar/weather differ
Only in voidrice-master-laggy/.local/bin: swapscreen
Only in voidrice-master-laggy/.local/bin: td-toggle
Only in voidrice-master-laggy/.local/bin: texclear
Only in voidrice-current-okay/.local/bin: tools
Only in voidrice-master-laggy/.local/bin: torwrap
Only in voidrice-master-laggy/.local/bin: transadd
Only in voidrice-current-okay/.local/bin: ttymaps.kmap
Only in voidrice-master-laggy/.local/bin: tutorialvids
Only in voidrice-master-laggy/.local/bin: unhide
Files voidrice-master-laggy/.local/bin/unix and voidrice-current-okay/.local/bin/unix differ
Only in voidrice-master-laggy/.local/bin: vifmimg
Only in voidrice-master-laggy/.local/bin: vu
Only in voidrice-master-laggy/.local/bin: wifi-check
Only in voidrice-master-laggy/.local/bin: wifi-restart
Only in voidrice-master-laggy/.local/bin: yt-playlist
Only in voidrice-master-laggy/.local/share/applications: file.desktop
Only in voidrice-master-laggy/.local/share/larbs: emoji
Only in voidrice-master-laggy/.local/share/larbs: getkeys
Files voidrice-master-laggy/.local/share/larbs/readme.mom and voidrice-current-okay/.local/share/larbs/readme.mom differ
Only in voidrice-master-laggy/.local/share/larbs: ttymaps.kmap
Files voidrice-master-laggy/.profile and voidrice-current-okay/.profile differ
Files voidrice-master-laggy/README.md and voidrice-current-okay/README.md differ
Files voidrice-master-laggy/.tmux.conf and voidrice-current-okay/.tmux.conf differ
Only in voidrice-current-okay: .vim
Only in voidrice-current-okay: .vimrc
Files voidrice-master-laggy/.xinitrc and voidrice-current-okay/.xinitrc differ
Files voidrice-master-laggy/.xprofile and voidrice-current-okay/.xprofile differ
Files voidrice-master-laggy/.yarnrc and voidrice-current-okay/.yarnrc differ
Files voidrice-master-laggy/.zprofile and voidrice-current-okay/.zprofile differ
Only in voidrice-current-okay: .zshrc
```

I'm pretty sure that the `Only in` stuff can be ignored since Luke changed up a few folders & the files exist, but in a different location.

---

Anything else I can help with?

**Comment by sarpik** [comment]
But now I just remembered that I've yet another USB available - I could try installing my own setup there, and checking if the issue persists.

Obviously there could be some things missing that wouldn't be included in the setup, like systemd services etc., but it's still worth a try I think. How do I install arch again? XD

**Comment by sarpik** [comment]
Hello there again, so I've updated once again to the latest changes from upstream (Luke's repo (this one)) and oh my god everything works fine again!!

I have no clue what fixed this exactly, but somehow everything's fine now 🤩

holy shit you guys made the colorful icons work by default in the terminal that's amazing

The only thing that was causing issues was that zsh took 500ms instead of 50ms to boot up. I fixed it by doing this:

```

**Comment by sarpik** [comment]
Other than that - huge thanks to both @LukeSmithxyz & @vladdoster - thanks for the help & everything else guys!
